tests/core/test_config.py

```diff
@@ -6,448 +6,510 @@
+    result, conflicts = config_merge(dest, source)
+
+    assert result == exp_result
+    assert conflicts == exp_conflicts
 
 
 @pytest.mark.parametrize(
-    "cfg_paths,contents,expected_exception,expected_log_entries",
+    "cfg_paths, expected_cfg_paths",
     [
-        (
-            ["Nonsense/file/location"],
-            [],
-            ConfigurationError,
+        ("string1", [Path("string1")]),
+        (Path("string1"), [Path("string1")]),
+        (["string1", "string2"], [Path("string1"), Path("string2")]),
+        (["string1", Path("string2")], [Path("string1"), Path("string2")]),
+        ([Path("string1"), Path("string2")], [Path("string1"), Path("string2")]),
+    ],
+)
+def test_Config_init(cfg_paths, expected_cfg_paths):
+    """Tests the normalisation of Config instance init."""
+    from virtual_rainforest.core.config import Config
+
+    # Just check normalisation, no processing
+    cfg = Config(cfg_paths, auto=False)
+
+    assert cfg.cfg_paths == expected_cfg_paths
+
+
+@pytest.mark.parametrize(
+    "cfg_paths,expected_exception,expected_log_entries",
+    [
+        pytest.param(
+            ["file_does_not_exist"],
+            pytest.raises(ConfigurationError),
             (
-                (
-                    CRITICAL,
-                    "The following (user provided) config paths do not exist:",
-                ),
+                (ERROR, "Config file path does not exist"),
+                (CRITICAL, "Config paths not all valid: check log."),
             ),
+            id="bad_path",
         ),
-        (
-            ["."],
-            [],
-            ConfigurationError,
+        pytest.param(
+            ["cfg_no_toml"],
+            pytest.raises(ConfigurationError),
             (
-                (
-                    CRITICAL,
-                    "The following (user provided) config folders do not contain any "
-                    "toml files:",
-                ),
+                (ERROR, "Config directory path contains no TOML files"),
+                (CRITICAL, "Config paths not all valid: check log."),
             ),
+            id="no_toml_dir",
         ),
-        (
-            ["", "all_config.toml"],
-            ["all_config.toml"],
-            ConfigurationError,
+        pytest.param(
+            ["bad_json_in_schema.json"],
+            pytest.raises(ConfigurationError),
             (
-                (
-                    CRITICAL,
-                    "A total of 1 config files are specified more than once (possibly "
-                    "indirectly)",
-                ),
+                (ERROR, "Config file path with non-TOML suffix"),
+                (CRITICAL, "Config paths not all valid: check log."),
+            ),
+            id="not_toml",
+        ),
+        pytest.param(
+            [".", "all_config.toml"],
+            pytest.raises(ConfigurationError),
+            (
+                (ERROR, "Repeated files in config paths:"),
```

Comment:
It's truncated. The `log_check` helper explicitly only validates the start of the string to avoid problems with changing paths etc. In this case, the real output is:

```
tests/core/test_config.py::test_Config_resolve_config_paths[dupes] [ERROR] - config - resolve_config_paths(376) - Repeated files in config paths: /private/var/folders/62/cz0ctby96lq7ggnytfbtd__40000gn/T/pytest-of-dorme/pytest-170/test_Config_resolve_config_pat3/data/all_config.toml
[CRITICAL] - config - resolve_config_paths(381) - Config paths not all valid: check log.
```

Reply:
Ahh right that makes sense (+ explains something I'd wondered about `log_check`)

Reply:
I think `log_check`  _has_ to work that way or you'll get run/platform specific errors.